### PR TITLE
[dv/otp_ctrl] add stress_all_test

### PR DESCRIPTION
**Pull request by cindychip**
This PR adds a few sequences to stress_all_test.
The chanllege for stress_all test is that otp_ctrl digest function
requires a reset to issue after it to make it work. Not following up
with a reset creates many corner cases on either RTL and DV side.

This PR moves a few variable to otp_ctrl_base_vseq to share among
all sequences within stress_all test.

Coming PRs will add more tests to stress_all.

Signed-off-by: Cindy Chen <chencindy@google.com>

### CODE REVIEW

File: hw/ip/otp_ctrl/dv/env/seq_lib/otp_ctrl_stress_all_vseq.sv [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      otp_ctrl_base_vseq otp_ctrl_vseq;
+      uint seq_idx = $urandom_range(0, seq_names.size - 1);
+
+      seq = create_seq_by_name(seq_names[seq_idx]);
+      `downcast(otp_ctrl_vseq, seq)
+
+      // if upper seq disables do_apply_reset for this seq, then can't issue reset
+      // as upper seq may drive reset
+      if (do_apply_reset) begin
+        otp_ctrl_vseq.do_apply_reset = $urandom_range(0, 1);
```
Comment (weicaiyang) [inline hw/ip/otp_ctrl/dv/env/seq_lib/otp_ctrl_stress_all_vseq.sv:31]: do you need to set `do_apply_reset=1` as you may need reset in sub-seq if upper-seq doesn't drive reset.

**Review by weicaiyang — APPROVED** [review]
LGTM with one question

File: hw/ip/otp_ctrl/dv/env/seq_lib/otp_ctrl_stress_all_vseq.sv (hunk 1 above)
Comment (cindychip) [reply hw/ip/otp_ctrl/dv/env/seq_lib/otp_ctrl_stress_all_vseq.sv:31]: Thanks for checking with me! Right now I add constraints on the sequence side to bypass the issue. The sequence so far works fine. Maybe need it once I add more sub-sequences to the stress_all sequence.

Comment (weicaiyang) [reply hw/ip/otp_ctrl/dv/env/seq_lib/otp_ctrl_stress_all_vseq.sv:31]: sg. Thanks

**Review by udinator — APPROVED** [review]
